storage/localstore/mode_set.go

```diff
@@ -393,3 +399,31 @@ func (db *DB) setUnpin(batch *leveldb.Batch, addr chunk.Address) (err error) {
 
 	return nil
 }
+
+// setReUpload adds a pinned chunk to the push index so that it can be re-uploaded to the network
+func (db *DB) setReUpload(batch *leveldb.Batch, addr chunk.Address) (err error) {
+	item := addressToItem(addr)
+
+	// only pinned chunks should be re-uploaded
+	// this also prevents a race condition: a non-pinned chunk could be gc'd after it's put in the push index
+	// but before it is actually re-uploaded to the network
+	_, err = db.pinIndex.Get(item)
+	if err != nil {
+		return err
```

Comment:
I see now that in tests, this error is checked as loveldb.ErrNotFound, but I think that it would be better if we would have more informative error chunk.ErrNotPinned which would be returned something like this:
```suggestion
                 if err == leveldb.ErrNotFound {
                     return chunk.ErrNotPinned
                 }
		return err
```

That would be reflected in the test, also.

Reply:
clearer this way indeed. thanks